module/zfs/dmu_objset.c

```diff
@@ -676,18 +695,26 @@ dmu_objset_refresh_ownership(objset_t *os, void *tag)
 {
 	dsl_pool_t *dp;
 	dsl_dataset_t *ds, *newds;
+	boolean_t key_needed;
 	char name[ZFS_MAX_DATASET_NAME_LEN];
 
 	ds = os->os_dsl_dataset;
 	VERIFY3P(ds, !=, NULL);
 	VERIFY3P(ds->ds_owner, ==, tag);
 	VERIFY(dsl_dataset_long_held(ds));
 
+	if (os->os_encrypted &&
+	    (spa_keystore_lookup_key(os->os_spa,
+	    os->os_dsl_dataset->ds_object, NULL, NULL) == 0))
+		key_needed = B_TRUE;
```

Comment:
I already have this fixed locally (and the other one as well). I will push it with the rest of these fixes.